src/librustc_metadata/decoder.rs

```diff
@@ -1147,6 +1148,22 @@ impl<'a, 'tcx> CrateMetadata {
         }
     }
 
+    pub fn defids_with_mir(&self, tcx: TyCtxt<'a, 'tcx, 'tcx>) -> DefIdSet {
+        if self.proc_macros.is_some() {
+            // Following the precedent of `exported_symbols()` we skip procedural macro crates.
+            DefIdSet::default()
+        } else {
+            let mut res = DefIdSet::default();
+            for idx in 1..self.root.entries_index.len - 1 {
```

Comment:
Why do we skip `idx` 0?

Reply:
Index zero is the crate itself, which won't have any MIR attached. Although the call to `tcx.is_mir_available()` would also perform that filtering. I don't mind making it `0..`.

Reply:
I would use whatever is most idiomatic: I have no opinion on this, but just wanted to flag it as something in case you hadn't fully thought about it.